cleanlab/count.py

```diff
@@ -263,16 +263,15 @@ def _compute_confident_joint_multi_label(
 
     # Compute unique number of classes K by flattening labels (list of lists)
     K = len(np.unique([i for lst in labels for i in lst]))
-    # Compute thresholds = p(label=k | k in set of given labels)
-    k_in_l = np.array([[k in lst for lst in labels] for k in range(K)])
     if thresholds is None:
         # the avg probability of class given that the label is represented.
-        thresholds = [np.mean(pred_probs[:, k][k_in_l[k]]) for k in range(K)]
+        thresholds = get_confident_thresholds(labels, pred_probs, multi_label=True)
     # Create mask for every example if for each class, prob >= threshold
     pred_probs_bool = pred_probs >= thresholds
 
     # Compute confident joint
     # (no need to avoid collisions for multi-label, double counting is okay!)
+    k_in_l = np.array([[k in lst for lst in labels] for k in range(K)])
```

Comment:
A nitpick is this code may be 2x as slow as the previous version for dataset with tons of labels, because you are computing this `k_in_l` object twice now (whereas previously it was only computed once), the first time being inside `get_confident_thresholds`

Reply:
I'm going to allow this redundancy for now. The two options to fix are:
1. compute `k_in_l` beforehand and pass it into `get_confident_thresholds`, but its a weird thing to pass in and the code is much easier to read if its computed inside of `get_confident_thresholds`
2. return `k_in_l` also when calling `get_confident_thresholds` with `multi_label=true` but that's pretty weird behavior. I can optionally include it in the return tuple with another parameter, but this seems like a somewhat silly amount of extra code to save very little compute.

the computation here should be relatively negligible so going to leave as is for the purpose of easier-to-read simpler code.

Reply:
Easy for me to implement either solution if you think its preferable. I vote to leave as is for code simplicity, but can go either way. Agree its 2x speed, but if we get to the place where this is the bottleneck than I'll be extremely happy :)